doc/debugging.rst

```diff
@@ -14,33 +14,108 @@ This allows breakpoints and the execution flow to be viewed and controlled right
 Debugging with ipdb
 ===================
 
-Spyder offers the following debugging features integrated into the native GUI:
+You can fully control debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` `console commands`_.
 
-* Multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+.. image:: images/debugging/debugging-commands.gif
+   :alt: A Spyder IPython console window, showing debugging from toolbar
 
-  * By selecting the respective option from the Debug menu.
-  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
-  * By double-clicking to the left of the line number in an open file.
-  * With an ``ipdb.set_trace()`` statement in your code (after ``import`` ing ``pdb``).
-  * Interactively, using the ``b`` command in an ``ipdb`` session.
+
+Additionally, the  :doc:`editor` shows the line of code the debugger is currently stopped on with an arrow.
+
+.. _console commands: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/
+
+.. image:: images/debugging/debugging-arrow.png
+   :alt: Spyder Editor showing the debugging panel
+
+
+Spyder's debugger offers syntax highlighting, code completion and command history, which work exactly like they do in the normal interactive interpreter.
+
+Use the up and down arrows to recall previous commands, and press :kbd:`Tab` to trigger autocomplete suggestions.
+
+.. image:: images/debugging/debugging-autocompletion.gif
```

Comment:
Not sure you need to show typing the `import numpy as np` line as part of the animated GIF unless typing it illustrates something I'm missing, just takes up extra time and space. Instead, start with it already entered, like you do with `li` in the next GIF.

Reply:
This one is fine for me, so it's up to you @juanis2112 if you want to follow @CAM-Gerlach's suggestion or not.

Reply:
This is addressed, thanks.